presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ApplyNode.java

```diff
@@ -62,19 +63,26 @@
      */
     private final Assignments subqueryAssignments;
 
+    /**
+     * Used for error reporting in case this ApplyNode is not supported
+     */
+    private final Node originSubquery;
+
     @JsonCreator
     public ApplyNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("input") PlanNode input,
             @JsonProperty("subquery") PlanNode subquery,
             @JsonProperty("subqueryAssignments") Assignments subqueryAssignments,
-            @JsonProperty("correlation") List<Symbol> correlation)
+            @JsonProperty("correlation") List<Symbol> correlation,
+            @JsonProperty("originSubquery") Node originSubquery)
```

Comment:
Also I think it is better to have a whole `Node` instead of just position:
 - `LateralJoinNode` and `ApplyNode` are about implementing subqueries so it is more natural a whole subquery there instead of just having its position
 - in case of debugging it is easier to understand what produced this node